File: user-service/src/main/java/com/athena/userservice/config/DataInitializer.java
package com.athena.userservice.config;

import com.athena.userservice.model.Group;
import com.athena.userservice.model.PasswordPolicy;
import com.athena.userservice.model.Role;
import com.athena.userservice.repository.GroupRepository;
import com.athena.userservice.repository.PasswordPolicyRepository;
import com.athena.userservice.repository.RoleRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
@Slf4j
public class DataInitializer implements CommandLineRunner {

    private final RoleRepository roleRepository;
    private final GroupRepository groupRepository;
    private final PasswordPolicyRepository passwordPolicyRepository;

    @Override
    public void run(String... args) {
        // Athena Credit Score roles
        createRoleIfNotFound("USER",        "Standard internal user");
        createRoleIfNotFound("ADMIN",       "Administrator with full access");
        createRoleIfNotFound("ANALYST",     "Credit analyst — view scores and reports");
        createRoleIfNotFound("VIEWER",      "Read-only access to dashboard");
        createRoleIfNotFound("CREDIT_RISK", "Credit risk officer — model configuration");

        // Athena Credit Score groups
        createGroupIfNotFound("ADMINISTRATORS",     "System administrators");
        createGroupIfNotFound("ANALYSTS",           "Credit analysts team");
        createGroupIfNotFound("CREDIT_RISK_TEAM",   "Credit risk management team");

        // Default password policy
        if (passwordPolicyRepository.count() == 0) {
            log.info("Seeding default password policy");
            passwordPolicyRepository.save(PasswordPolicy.builder()
                    .minLength(8)
                    .requireUppercase(true)
                    .requireLowercase(true)
                    .requireNumbers(true)
                    .requireSpecialChars(false)
                    .expirationDays(90)
                    .build());
        }
    }

    private Role createRoleIfNotFound(String name, String description) {
        return roleRepository.findByName(name).orElseGet(() -> {
            log.info("Seeding role: {}", name);
            return roleRepository.save(Role.builder().name(name).description(description).build());
        });
    }

    private Group createGroupIfNotFound(String name, String description) {
        return groupRepository.findByName(name).orElseGet(() -> {
            log.info("Seeding group: {}", name);
            return groupRepository.save(Group.builder().name(name).description(description).build());
        });
    }
}

File: user-service/internal/seed/data_initializer.go
package seed

import (
	"github.com/athena/user-service/internal/model"
	"github.com/athena/user-service/internal/repository"
	"github.com/rs/zerolog/log"
)

// Initialize seeds roles, groups, and default password policy.
func Initialize(
	roleRepo *repository.RoleRepository,
	groupRepo *repository.GroupRepository,
	policyRepo *repository.PasswordPolicyRepository,
) {
	// Athena Credit Score roles
	createRoleIfNotFound(roleRepo, "USER", "Standard internal user")
	createRoleIfNotFound(roleRepo, "ADMIN", "Administrator with full access")
	createRoleIfNotFound(roleRepo, "ANALYST", "Credit analyst — view scores and reports")
	createRoleIfNotFound(roleRepo, "VIEWER", "Read-only access to dashboard")
	createRoleIfNotFound(roleRepo, "CREDIT_RISK", "Credit risk officer — model configuration")

	// Athena Credit Score groups
	createGroupIfNotFound(groupRepo, "ADMINISTRATORS", "System administrators")
	createGroupIfNotFound(groupRepo, "ANALYSTS", "Credit analysts team")
	createGroupIfNotFound(groupRepo, "CREDIT_RISK_TEAM", "Credit risk management team")

	// Default password policy
	if policyRepo.Count() == 0 {
		log.Info().Msg("Seeding default password policy")
		policy := &model.PasswordPolicy{
			MinLength:          8,
			RequireUppercase:   true,
			RequireLowercase:   true,
			RequireNumbers:     true,
			RequireSpecialChars: false,
			ExpirationDays:     90,
		}
		if err := policyRepo.Create(policy); err != nil {
			log.Error().Err(err).Msg("failed to seed password policy")
		}
	}
}

func createRoleIfNotFound(repo *repository.RoleRepository, name, description string) {
	if _, err := repo.FindByName(name); err != nil {
		log.Info().Str("role", name).Msg("Seeding role")
		role := &model.Role{Name: name, Description: description}
		if err := repo.Create(role); err != nil {
			log.Error().Err(err).Str("role", name).Msg("failed to seed role")
		}
	}
}

func createGroupIfNotFound(repo *repository.GroupRepository, name, description string) {
	if _, err := repo.FindByName(name); err != nil {
		log.Info().Str("group", name).Msg("Seeding group")
		group := &model.Group{Name: name, Description: description}
		if err := repo.Create(group); err != nil {
			log.Error().Err(err).Str("group", name).Msg("failed to seed group")
		}
	}
}
